examples/oneapi/cpp/source/svc/svm_thunder_dense_batch.cpp

```diff
@@ -0,0 +1,113 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <iomanip>
+#include <iostream>
+
+#include "oneapi/dal/data/table.hpp"
+#include "oneapi/dal/data/accessor.hpp"
+#include "oneapi/dal/algo/svm.hpp"
+
+using namespace oneapi;
+
+std::ostream &operator <<(std::ostream& stream, const dal::table& table) {
+    auto arr = dal::row_accessor<const float>(table).pull();
+    const auto x = arr.get_data();
+
+    for (std::int64_t i = 0; i < table.get_row_count(); i++) {
+        for (std::int64_t j = 0; j < table.get_column_count(); j++) {
+            std::cout << std::setw(10)
+                      << std::setiosflags(std::ios::fixed)
+                      << std::setprecision(3)
+                      << x[i * table.get_column_count() + j];
+        }
+        std::cout << std::endl;
+    }
+    return stream;
+}
+
+onedal::svm::model run_training(const onedal::table& x_table,
+                                const onedal::table& y_table) {
+
+    const auto svm_desc = dal::svm::descriptor<float, task::classification, method::thunder>{}
+        .set_c(3.0)
+        // .set_kernel(NAN) is TODO (?)
+        .set_accuracy_threshold(0.01)
+        .set_max_iteration_count(1000)
+        .set_cache_size(8) // In GB/MB/B (?)
+        .set_tau(0.01);
+
+   const auto result = onedal::train(svm_desc, x_table, y_table);
+   return result.get_model();
+}
+
+onedal::table run_inference(const onedal::table& x_table,
+                                const onedal::table& y_table) {
+
+    const auto svm_desc = dal::svm::descriptor<float, task::classification>{}
```

Comment:
I'd suggest to reuse descriptor: the same descriptor for both training and inference

Reply:
But in predict can use other method. For example: multi class have two method: `wu`, `vote` and training have only method: `oneAgainstOne`. Some situation for SVM, training have two method: `thunder` and `boser` and predict have only `DefaultDense`. By the way, I’m thinking of renaming methods for training SVM: `boser` -> `smo`, `thunder` -> `block_smo`. What do you think?